CMakeLists.txt

```diff
@@ -33,6 +38,7 @@ add_compile_options($<$<CONFIG:Release>:-Wuninitialized>)
 add_compile_options($<$<CONFIG:Release>:-Wno-dangling-else>)
 add_compile_options(-Wstack-protector -fstack-protector)
 add_compile_options(-Wstrict-aliasing -fstrict-aliasing)
+add_compile_options(-D_GNU_SOURCE)
```

Comment:
Please consider adding them conditionally using `target_compile_definitions`. It feels strange to see `-D_GNU_SOURCE` on command-line when working on FreeBSD.